src/result.c

```diff
@@ -407,21 +423,47 @@ bool result_put(const char *path, struct filelist *list, int compression_level)
 		goto out;
 	}
 
+	uint8_t compr_type;
+	if (compression_level == 0) {
+		compr_type = COMPR_TYPE_NONE;
+	} else {
+#ifdef HAVE_LIBZSTD
+		compr_type = COMPR_TYPE_ZSTD;
+#else
+		compr_type = COMPR_TYPE_ZLIB;
+#endif
+	}
+
 	char header[7];
 	memcpy(header, MAGIC, sizeof(MAGIC));
 	header[4] = RESULT_VERSION;
-	header[5] = compression_level == 0 ? COMPR_TYPE_NONE : COMPR_TYPE_ZLIB;
+	header[5] = compr_type;
 	header[6] = compression_level;
 	if (fwrite(header, 1, sizeof(header), f) != sizeof(header)) {
 		cc_log("Failed to write to %s", tmp_file);
 		goto out;
 	}
 
 	struct compressor *compressor;
-	if (compression_level == 0) {
+	switch (compr_type) {
+	case COMPR_TYPE_NONE:
 		compressor = &compr_none;
-	} else {
+		break;
+
+	case COMPR_TYPE_ZLIB:
 		compressor = &compr_zlib;
+		break;
+
+#ifdef HAVE_LIBZSTD
+	case COMPR_TYPE_ZSTD:
+		ZWRAP_useZSTDcompression(1);
```

Comment:
You need to support variable compression levels here.

Reply:
The levels are still handled by the "zlib" compressor.

It is unfortunate that they have different ranges, but.